homeassistant/components/bluetooth/update_coordinator.py

```diff
@@ -70,7 +69,7 @@ def name(self) -> str:
         if service_info := async_last_service_info(
             self.hass, self.address, self.connectable
         ):
-            return cast(str, service_info.name)  # for compat this can be a pyobjc
+            return service_info.name
```

Comment:
The typing isn't `Any` now in the upstream lib so python can see it but since its typed as `_str` cython will keep it a python object and won't try to optimize it so we get the best of both